cdap-common/bin/functions.sh

```diff
@@ -503,15 +503,31 @@ cdap_set_spark() {
       # If there is no valid SPARK_HOME, we should unset the existing one if it is set
       # Otherwise the spark-shell won't run correctly
       unset SPARK_HOME
+      local __spark_shell=$(which spark-shell)
+      local __spark_client_version=None
+      for __dist in hdp iop; do
```

Comment:
how does this work on CDH packages?  pretty sure it will hit the last `else` below and fail to detect spark